gitops/predicates.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitops
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// DeploymentStatusChangedPredicate returns a predicate which filters out update events to a
+// SnapshotEnvironmentBinding where the component deployment status goes from unknown to true/false.
+func DeploymentStatusChangedPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isDeploymentStatusKnown(e.ObjectOld, e.ObjectNew)
```

Comment:
Same here. `hasDeploymentFinished` is more precise. In the case of the tekton predicate the name is not good enough either but that we can change it any other time.